File: pset6/vigenere/vigenere.py
import sys
from cs50 import get_string

if not len(sys.argv) == 2:
    print("Usage: key")
    sys.exit(1)

#User provided encryption key via command line argument
key = sys.argv[1]

#Key must be alphabetic
for c in key:
    if not c.isalpha():
        print("Usage: key")
        sys.exit(1)

#Prompt user for text to be encrypted
plaintext = get_string("plaintext: ")
i = 0

print("ciphertext: ", end="")

for c in plaintext:
    #Checks for non-alphabetic input and outputs without changes
    if not c.isalpha():
        print(c, end="")
        continue

    #Takes alphabetic input, preserves case, encrypts, and outputs ciphertext
    ascii_offset = 65 if c.isupper() else 97
    p_index = ord(c) - ascii_offset
    key_index = ord(key[i % len(key)].upper()) - 65
    cipher = (p_index + key_index) % 26
    i += 1

    print(chr(cipher + ascii_offset), end="")

print()

File: pset2/vigenere/vigenere.c
#include<stdio.h>
#include<cs50.h>
#include<string.h>
#include<stdlib.h>
#include<ctype.h>

int main(int argc, string argv[])
{
    //Conditions for command line argument
    if (argc != 2)
    {
        printf("Usage: ./vigenere key\n");
        return 1;
    }

    for (int i = 0, n = strlen(argv[1]); i < n; i++)
    {
        if (! isalpha(argv[1][i]))
        {
            printf("Usage: ./vigenere key\n");
            return 1;
        }
    }

    //Prompt the user for text that needs to be encrypted
    string plaintext = get_string("plaintext: ");

    printf("ciphertext: ");

    int key[strlen(argv[1])];

    //Pass user supplied keyword into an int array and shift from ascii value to index value
    for (int i = 0, n = strlen(argv[1]); i < n; i++)
    {
        if (isupper(argv[1][i]))
        {
            key[i] = argv[1][i] - 'A';
        }
        else if (islower(argv[1][i]))
        {
            key[i] = argv[1][i] - 'a';
        }
    }

    //Uses the new key to encrypt plaintext
    for (int i = 0, j = 0, n = strlen(plaintext); i < n; i++)
    {
        if (isalpha(plaintext[i]))
        {
            if (isupper(plaintext[i]))
            {
                printf("%c", (plaintext[i] + key[j % strlen(argv[1])] - 'A') % 26 + 'A');
            }
            else if (islower(plaintext[i]))
            {
                printf("%c", (plaintext[i] + key[j % strlen(argv[1])] - 'a') % 26 + 'a');
            }
            j++;
        }
        else
        {
            printf("%c", plaintext[i]);
        }
    }
    printf("\n");
}